packages/editor-ui/src/components/__tests__/RunData.test.ts

```diff
@@ -6,30 +6,87 @@ import RunData from '@/components/RunData.vue';
 import { STORES, VIEWS } from '@/constants';
 import { SETTINGS_STORE_DEFAULT_STATE } from '@/__tests__/utils';
 import { createComponentRenderer } from '@/__tests__/render';
-
-const renderComponent = createComponentRenderer(RunData, {
-	props: {
-		nodeUi: {
-			name: 'Test Node',
-		},
-	},
-	data() {
-		return {
-			canPinData: true,
-		};
-	},
-	global: {
-		mocks: {
-			$route: {
-				name: VIEWS.WORKFLOW,
-			},
-		},
-	},
-});
+import type { IRunDataDisplayMode } from '@/Interface';
 
 describe('RunData', () => {
 	it('should render data correctly even when "item.json" has another "json" key', async () => {
-		const { html, getByText, getAllByTestId, getByTestId } = renderComponent({
+		const { getByText, getAllByTestId, getByTestId } = render(
+			[
+				{
+					json: {
+						id: 1,
+						name: 'Test 1',
+						json: {
+							data: 'Json data 1',
+						},
+					},
+				},
+				{
+					json: {
+						id: 2,
+						name: 'Test 2',
+						json: {
+							data: 'Json data 2',
+						},
+					},
+				},
+			],
+			'schema',
+		);
+
+		await userEvent.click(getByTestId('ndv-pin-data'));
+		await waitFor(() => getAllByTestId('run-data-schema-item'), { timeout: 1000 });
+		expect(getByText('Test 1')).toBeInTheDocument();
+		expect(getByText('Json data 1')).toBeInTheDocument();
+	});
+
+	it('should render view and download options for PDFs', async () => {
+		const { getByTestId } = render(
+			[
+				{
+					json: {},
+					binary: {
+						data: {
+							data: 'filesystem',
+							id: 'filesystem:12110d8f95e1c-c64f-4dbe-8ea6-a5282d957737',
+							directory: 'somewhere',
+							fileExtension: 'pdf',
+							fileName: 'test.pdf',
+							fileSize: '11.5 kB',
+							fileType: 'pdf',
+							mimeType: 'application/pdf',
+						},
+					},
+				},
+			],
+			'binary',
+		);
+		// await userEvent.click(getByTestId('radio-button-binary'));
+		const binaryData = getByTestId('ndv-binary-data_0');
+		binaryData.querySelector('a');
```

Comment:
should this be asserting something?

Reply:
ugh. I was so excited about having fixed the "issue", I forgot to commit the assertion 😅